Fixes Issue#443

Supriya1702 opened the pull request:
<!-- Please be sure to check out our contributing guidelines,
https://github.com/radis/radis/blob/develop/CONTRIBUTING.md . -->

### Description
<!-- Provide a general description of what your pull request does. -->

This PR does the following:
- If `truncation=None` for `broadening_method='voigt'` for `optimization='simple' or 'min-RMS` method it raises NotImplementedError
- If `truncation` is negative it raises value error
- Allows computation for `broadening_method == "voigt"` and `truncation == None` for `optimization=None`

<!-- If the pull request closes any open issues you can add this.
If you replace <Issue Number> with a number, GitHub will automatically link it.
If this pull request is unrelated to any issues, please remove
the following line. -->

Fixes #443

anandxkumar commented:
Tested for different cases, works fine on my end, Good job! 
@erwanp should `truncation=None` work for `optimization=None` for `broadening_method='voigt'`?

erwanp commented on radis/lbl/factory.py:
```diff
@@ -542,7 +542,13 @@ def __init__(
         if truncation == 0:
             raise ValueError(
-                "Lineshape truncation must be >0. If you want no truncation (compute lineshape on the full spectral range), use `truncation=None`"
+                "Lineshape truncation must be > 0. If you want no truncation (compute lineshape on the full spectral range), use `truncation=None`"
```
Can you also add :

"Note (advanced) : no truncation is not physically more accurate. Most molecules exhibit a sub-lorentzian behavior far from the line centers. A truncation at around 40-50 cm-1 is a good choice"

erwanp reviewed (changes requested):
Hello, good work! 

1. are you sure optimization=None is implemented?

2. We'd need a quick test, for instance one making sure an error is raised if negative truncation is given, and one making sure truncation=None does what we want. If you need help implementing tests please reach out on Slack

Supriya1702 commented:
Hello @erwanp, I added test for checking negative truncation and `truncation==None`. I found that `truncation=None` works for `broadening_method='voigt'` for `optimization=None` but gives error when used `fft`. Oppositely in `optimization='simple`, `truncation=None` works for `fft` but fails for `voigt`, so I added the tests accordingly
